exporter/elasticsearchexporter/partitioner.go

```diff
@@ -38,3 +40,33 @@ func (p metadataKeysPartitioner) GetKey(
 	}
 	return kb.String()
 }
+
+func (p metadataKeysPartitioner) MergeCtx(
+	ctx1, ctx2 context.Context,
+) context.Context {
+	m1 := client.FromContext(ctx1).Metadata
+	m2 := client.FromContext(ctx2).Metadata
+
+	m := make(map[string][]string, len(p.keys))
+	for _, key := range p.keys {
+		v1 := m1.Get(key)
+		v2 := m2.Get(key)
+		if len(v1) == 0 && len(v2) == 0 {
+			continue
+		}
+
+		// Since the mergeCtx is based on partition key, we MUST have the same
+		// partition key-values in both the metadata. If they are not same then
+		// fail fast and dramatically.
+		if !slices.Equal(v1, v2) {
+			panic(fmt.Errorf(
+				"unexpected client metadata found when merging context for key %s", key,
+			))
+		}
+		m[key] = v1
+	}
+	return client.NewContext(
+		context.Background(),
```

Comment:
🤔 is there definitely nothing else in the contexts that we should propagate? Would it make sense to use either one of ctx1/ctx2 as the base, and only override the client info?

Reply:
I was a bit hesitant here as well, but collector code handles the timing propagation. Haven't checked cancellation